lib/demos/reference/transformations_demo.dart

```diff
@@ -17,29 +18,91 @@ class TransformationsDemo extends StatefulWidget {
   _TransformationsDemoState createState() => _TransformationsDemoState();
 }
 
-class _TransformationsDemoState extends State<TransformationsDemo> {
+class _TransformationsDemoState extends State<TransformationsDemo>
+    with TickerProviderStateMixin {
   // The radius of a hexagon tile in pixels.
   static const _kHexagonRadius = 32.0;
   // The margin between hexagons.
   static const _kHexagonMargin = 1.0;
   // The radius of the entire board in hexagons, not including the center.
   static const _kBoardRadius = 12;
 
-  bool _reset = false;
   Board _board = Board(
     boardRadius: _kBoardRadius,
     hexagonRadius: _kHexagonRadius,
     hexagonMargin: _kHexagonMargin,
   );
 
+  bool _firstRender = true;
+  Matrix4 _homeTransformation;
+  final TransformationController _transformationController =
+      TransformationController();
+  Animation<Matrix4> _animationReset;
+  AnimationController _controllerReset;
+
+  // Handle reset to home transform animation.
+  void _onAnimateReset() {
+    _transformationController.value = _animationReset.value;
+    if (!_controllerReset.isAnimating) {
+      _animationReset?.removeListener(_onAnimateReset);
+      _animationReset = null;
+      _controllerReset.reset();
+    }
+  }
+
+  // Initialize the reset to home transform animation.
+  void _animateResetInitialize() {
+    _controllerReset.reset();
+    _animationReset = Matrix4Tween(
+      begin: _transformationController.value,
+      end: _homeTransformation,
+    ).animate(_controllerReset);
+    _controllerReset.duration = const Duration(milliseconds: 400);
+    _animationReset.addListener(_onAnimateReset);
+    _controllerReset.forward();
+  }
+
+  // Stop a running reset to home transform animation.
+  void _animateResetStop() {
+    _controllerReset.stop();
+    _animationReset?.removeListener(_onAnimateReset);
+    _animationReset = null;
+    _controllerReset.reset();
+  }
+
+  void _onScaleStart(ScaleStartDetails details) {
+    // If the user tries to cause a transformation while the reset animation is
+    // running, cancel the reset animation.
+    if (_controllerReset.status == AnimationStatus.forward) {
+      _animateResetStop();
+    }
+  }
+
+  void _onTapUp(TapUpDetails details) {
+    final scenePoint = _transformationController.toScene(details.localPosition);
+    final boardPoint = _board.pointToBoardPoint(scenePoint);
+    setState(() {
+      _board = _board.copyWithSelected(boardPoint);
+    });
+  }
+
   @override
-  Widget build(BuildContext context) {
-    final painter = BoardPainter(
-      board: _board,
+  void initState() {
+    super.initState();
+    _controllerReset = AnimationController(
+      vsync: this,
     );
+  }
 
+  @override
+  void didUpdateWidget(TransformationsDemo oldWidget) {
```

Comment:
Do you need this since all it's doing is calling super?

Reply:
Nope, good catch.